src/Auth.php

```diff
@@ -50,6 +50,27 @@ public function withHandler(string $handler = null)
 		return $this;
```

Comment:
I know this was a previous commit, but using a "with____" method to me implies an immutable class. I would recommend either returning a cloned version or using `setHandler` instead.

Reply:
Fair point, and since I don't see the point in immutable classes for many things, I'm happy to change the method name.

Reply:
In this case I think a setter works great. I've been warming to immutability. I think in general PHP just isn't a language where it will make too much difference, but I've been poking around at Swoole and the concurrency gains are pretty impressive. It also seems to be a growing trend - whether it ends up being a fad or "here to stay" remains to be seen but I've been incorporating more immutable classes in my work when it makes sense in the name of collaboration.

Reply:
I know it’s a trend and I think in many of the cases I’ve seen it used there was no gain - and a number of extra copies of that data in memory. But that could be the old grumpy man talking lol

I hadn’t heard of Swoole before but I’m definitely intrigued and will be checking it out more soon. Could prove very useful for a personal project I’m in the beginning stages of....

Reply:
Oh yeah definitely look at Swoole! I think the whole thing typically with immutability in PHP is that our program execution time, start to finish, is often mere milliseconds. Swoole and other such extensions introduce long-running processes, so all of a sudden things that didn't matter before can make or break an app. That said, doing things "the right way" can also really pay off. At the end of the day my rubric is still something like "does this class *make sense conceptually* as an immutable object"... if it is going to make my code easier to grasp then that's great, if it doesn't matter then I'll try immutability for the practice at this point, if it makes things super convoluted then screw that :P